code/src/pages/dashboard/dashboard-products/components/dashboard-products-header/DashboardProductsHeader.tsx

```diff
@@ -0,0 +1,65 @@
+import { ChangeEvent, useState } from "react";
+import { useIntl } from "react-intl";
+
+import AddIcon from "@mui/icons-material/Add";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppButton from "@/components/app-button/AppButton";
+import AppSearchInput from "@/components/app-search-input/AppSearchInput";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import routes from "@/constants/routes";
+import { DashboardProductsHeaderProps } from "@/pages/dashboard/dashboard-products/DashboardProductsPage.types";
+
+const DashboardProductsHeader = ({
+  onSearch
+}: DashboardProductsHeaderProps) => {
+  const [searchValue, setSearchValue] = useState("");
+
+  const { formatMessage } = useIntl();
+
+  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setSearchValue(value);
+  };
+
+  const handleClearSearch = () => setSearchValue("");
+
+  const handleSearch = () => {
```

Comment:
probably you will fased with the problem, that when you are not on the first page and ckick on handleSearch button, the table would be empty, because results are on the previous page
My deсision was change also page param into 1, to redirect user on first list page

Reply:
Thank you for notice. I haven't encountered such problem. I think that is because in your case hook for filtering is used. I will also check behaviour with real backend in few minutes, when it will be available.

Reply:
I checked it with real backend, everything works fine